tests/distributions/test_one_one_matching.py

```diff
@@ -0,0 +1,116 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import math
+
+import pytest
+import torch
+
+import pyro.distributions as dist
+from tests.common import assert_close, assert_equal, xfail_if_not_implemented
+
+BP_ITERS = 30
+
+
+def _hash(value):
+    return tuple(value.tolist())
+
+
+@pytest.mark.parametrize("dtype", [torch.float, torch.double], ids=str)
+@pytest.mark.parametrize("num_nodes", [1, 2, 3, 4, 5, 6])
+def test_enumerate(num_nodes, dtype):
+    logits = torch.randn(num_nodes, num_nodes, dtype=dtype)
+    d = dist.OneOneMatching(logits)
+    values = d.enumerate_support()
+    logging.info("destins = {}, suport size = {}".format(num_nodes, len(values)))
+    assert d.support.check(values), "invalid"
+    assert len(set(map(_hash, values))) == len(values), "not unique"
+
+
+@pytest.mark.parametrize("dtype", [torch.float, torch.double], ids=str)
+@pytest.mark.parametrize("sample_shape", [(), (4,), (3, 2)], ids=str)
+@pytest.mark.parametrize("num_nodes", [1, 2, 3, 4, 5, 6])
+@pytest.mark.parametrize("bp_iters", [None, BP_ITERS], ids=["exact", "bp"])
+def test_sample_shape_smoke(num_nodes, sample_shape, dtype, bp_iters):
+    logits = torch.randn(num_nodes, num_nodes, dtype=dtype)
+    d = dist.OneOneMatching(logits, bp_iters=bp_iters)
+    with xfail_if_not_implemented():
+        values = d.sample(sample_shape)
+    assert values.shape == sample_shape + (num_nodes,)
+    assert d.support.check(values).all()
+
+
+@pytest.mark.parametrize("dtype", [torch.float, torch.double], ids=str)
+@pytest.mark.parametrize("num_nodes", [1, 2, 3, 4, 5, 6, 7, 8])
+@pytest.mark.parametrize("bp_iters", [None, BP_ITERS], ids=["exact", "bp"])
+def test_log_prob_full(num_nodes, dtype, bp_iters):
```

Comment:
do you think it would be useful to pair one of these with a finite difference gradient test?

Reply:
good idea. I've added gradient check tests for both `OneOneMatching` and `OneTwoMatching`. Note the tolerances are pretty coarse (atol=rtol=0.01) and I needed to run for a few more iterations.